.github/workflows/check_tests.yml

```
@@ -0,0 +1,17 @@
+name: Check Tests
+on: [ pull_request ]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'  # Specify the Python version you want to use
```

```suggestion
          python-version: '3.x'  # Specify the Python version you want to use
          cache: pip
```
This will enable caching for pip dependencies.

---

Thanks, added!